lib/src/http_client.dart

```diff
@@ -0,0 +1,281 @@
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> get(String host, int port, String path) {
+    return getUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using HEAD method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> headUrl(Uri url) {
+    return openUrl('HEAD', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using HEAD method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> head(String host, int port, String path) {
+    return headUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using PUT method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> putUrl(Uri url) {
+    return openUrl('PUT', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using PUT method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> put(String host, int port, String path) {
+    return putUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using POST method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> postUrl(Uri url) {
+    return openUrl('POST', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using POST method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> post(String host, int port, String path) {
+    return postUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using PATCH method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> patchUrl(Uri url) {
+    return openUrl('PATCH', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using PATCH method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> patch(String host, int port, String path) {
+    return patchUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using DELETE method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> deleteUrl(Uri url) {
+    return openUrl('DELETE', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using DELETE method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> delete(String host, int port, String path) {
+    return deleteUrl(_getUri(host, port, path));
+  }
+
+  /// Function for resolving the proxy server to be used for a HTTP connection from
+  /// the proxy configuration specified through environment variables.
+  ///
+  /// Note: It just returns `dart:io` [HttpClient.findProxyFromEnvironment].
+  static String findProxyFromEnvironment(Uri url,
+      {Map<String, String>? environment}) {
+    return io.HttpClient.findProxyFromEnvironment(url,
+        environment: environment);
+  }
+
+  /// Gets Cronet's version
```

Comment:
With getters/setters, we don't use a verb in Dart for the documentation. We just describe what it is rather than using a verb for 'getting' and 'setting'.

https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases

(Also, missing period at the end of the sentence.)